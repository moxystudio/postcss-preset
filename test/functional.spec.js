'use strict';

const path = require('path');
const fs = require('fs');
const postcss = require('postcss');
const preset = require('..');

const postcssInstance = (() => {
    const { plugins } = preset({
        url: { url: 'inline' },
    });

    const instance = postcss();

    plugins.forEach((plugin) => instance.use(plugin));

    return instance;
})();

const getOutput = (file) => {
    const input = fs.readFileSync(file);

    return postcssInstance.process(input, { from: file });
};

const fixtureFiles = [
    'fixtures/example.css',
];

fixtureFiles.forEach((fixtureFile) => {
    it(`should generate correct output for ${fixtureFile}`, async () => {
        const output = await getOutput(path.join(__dirname, fixtureFile));

        expect(output.css).toMatchSnapshot();
    });
});
