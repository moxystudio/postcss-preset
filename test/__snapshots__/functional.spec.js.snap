// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate correct output for fixtures/example.css 1`] = `
".image {
    -webkit-transition: all 0.2s ease;
    transition: all 0.2s ease;
    font-family: Verdana;
    color: red;
    background: url('data:image/png;base64,');
    background-image: -webkit-image-set(url(\\"data:image/png;base64,\\") 1x, url(\\"data:image/png;base64,\\") 2x);
    background-image: image-set(url(\\"data:image/png;base64,\\") 1x, url(\\"data:image/png;base64,\\") 2x);
    padding: calc(20px * 2);
        border-color: blue
}



@media (min-width: 60px) {



    .image {
    padding: calc(30px * 2)
    }
}



.image .bar a {
            font-family: Verdana;
            color: red;
            background: url('data:image/png;base64,');
            background-image: -webkit-image-set(url(\\"data:image/png;base64,\\") 1x, url('data:image/png;base64,') 2x);
            background-image: image-set(url(\\"data:image/png;base64,\\") 1x, url('data:image/png;base64,') 2x);
            padding: calc(20px * 2);
                border-color: blue
        }



@media (min-width: 60px) {



    .image .bar a {
            padding: calc(30px * 2)
    }
}



.image .bar .link {
            font-family: Verdana;
            color: red;
            background: url('data:image/png;base64,');
            background-image: -webkit-image-set(url(\\"data:image/png;base64,\\") 1x, url('data:image/png;base64,') 2x);
            background-image: image-set(url(\\"data:image/png;base64,\\") 1x, url('data:image/png;base64,') 2x);
            padding: calc(20px * 2);
                border-color: blue
        }



@media (min-width: 60px) {



    .image .bar .link {
            padding: calc(30px * 2)
    }
}



.image .bar a:hover {
                font-family: Verdana;
                color: red;
                background: url('data:image/png;base64,');
                background-image: -webkit-image-set(url(\\"data:image/png;base64,\\") 1x, url(\\"data:image/png;base64,\\") 2x);
                background-image: image-set(url(\\"data:image/png;base64,\\") 1x, url(\\"data:image/png;base64,\\") 2x);
                padding: calc(20px * 2);
                    border-color: blue
            }



@media (min-width: 60px) {



    .image .bar a:hover {
                padding: calc(30px * 2)
    }
}



.image .bar .link:hover {
                font-family: Verdana;
                color: red;
                background: url('data:image/png;base64,');
                background-image: -webkit-image-set(url(\\"data:image/png;base64,\\") 1x, url(\\"data:image/png;base64,\\") 2x);
                background-image: image-set(url(\\"data:image/png;base64,\\") 1x, url(\\"data:image/png;base64,\\") 2x);
                padding: calc(20px * 2);
                    border-color: blue
            }



@media (min-width: 60px) {



    .image .bar .link:hover {
                padding: calc(30px * 2)
    }
}



.buttonWrapper .button.tertiary:enabled:hover:not(.loading) span {
                    color: rgba(0, 0, 0, 0.8);
                }
"
`;
